server/server.js

```diff
@@ -138,4 +137,92 @@ app.get('/sidebar/all', (req, res) => {
   });
 });
 
+// Create
+app.post('/sidebar/all', async (req, res) => {
+  console.log('POST request to /sidebar/all');
+  const newDocument = req.body;
+  const newCourseId = newDocument.courseId;
+  console.log(newDocument);
+  // change string date to date type
+  newDocument.price.saleEndDate = new Date(newDocument.price.saleEndDate);
+
+  if (typeof newCourseId !== 'number') {
+    res.status(400).send('Sorry, invalid request: courseId is not a number');
+  } else {
+    db.getSidebar({ courseId: newCourseId }, async (err, docs) => {
+      console.log('return from courseId query:', docs);
+      if (err) {
+        console.warn('Error Occured :', err);
+      } else if (docs[0] === undefined) {
+        console.log('courseId is available!');
+        await db.postAll(newDocument)
+          .then((result) => {
+            console.log(result);
+            res.status(201).send(result);
+          })
+          .catch((error) => {
+            console.warn('Error occured during create (server side): ', error);
+          });
+      } else {
+        res.status(400).send('Sorry, courseId already exists.');
+      }
+    });
+  }
+});
+
+// Delete
+app.delete('/sidebar/all', async (req, res) => {
+  console.log('DELETE request to /sidebar/all', req.query);
+  const courseId = req.query;
+  await db.deleteAll(courseId)
+    .then((result) => {
+      if (result) {
+        res.status(204);
+      } else {
+        throw Error('Check DB side');
+      }
+      res.end();
+    })
+    .catch((error) => {
+      console.warn('Error occured during delete', error);
+      res.status(400).send('Sorry, Error in deleting occured.');
+      res.end();
+    });
+});
+
+// Update
+app.put('/sidebar/all', async (req, res) => {
```

Comment:
I would recommend to replace '/sidebar/all' differently for each individual CRUD API call. You can refer GLearn, which has an example for that.

Reply:
so I wrote the API endpoint in a way which I wouldn't have to write three endpoints for an update. In talking with Leslie at office hours, she said that this was fine as long as I wrote the code within in a way that would be able to parse the information being updated properly, which is what I have done. I agree that it might not be as clean, but in an effort to avoid spending time on redundant tasks and get the "most out of this project" (per my OH with Leslie) I went this way. But I do see what you are saying -- to make 3 separate CRUD API's for Price, Sidebar and Preview Video.